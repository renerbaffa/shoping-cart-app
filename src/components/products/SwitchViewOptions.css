.SwitchViewOptions-button {
  height: 31px;
}
