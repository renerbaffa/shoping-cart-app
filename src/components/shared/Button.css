.Button {
  background-color: white;
  padding: 0px 18px;
}
