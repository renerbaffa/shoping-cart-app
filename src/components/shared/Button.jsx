import React from 'react';

/*
  Button component is an adapter.
  Imagine that in the future we move to react-bootstrap or material-ui
  In this case we only need to replase default button by the library button
*/
const Button = ({ children, ...other }) => (
  <button {...other}>
    {children}
  </button>
);

export default Button;
