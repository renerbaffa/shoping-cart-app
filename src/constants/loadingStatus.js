export const FAILED = 'FAILED';
export const RETRIEVED = 'RETRIEVED';
export const RETRIEVING = 'RETRIEVING';
