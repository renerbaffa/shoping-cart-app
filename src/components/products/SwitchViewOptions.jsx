import React from 'react';
import PropTypes from 'prop-types';

import Button from '../shared/Button';

import { GRID, LIST } from '../../constants/ViewOptions';

import './SwitchViewOptions.css';

const SwitchViewOptions = ({ currentView, onSwitchView, ...other }) => (
  <div {...other}>
    <Button
      className="SwitchViewOptions-button"
      selected={currentView === GRID}
      onClick={() => onSwitchView(GRID)}
    >
      Grid
    </Button>
    <Button
      className="SwitchViewOptions-button"
      selected={currentView === LIST}
      onClick={() => onSwitchView(LIST)}
    >
      List
    </Button>
  </div>
);

SwitchViewOptions.propTypes = {
  currentView: PropTypes.oneOf([GRID, LIST]),
  onSwitchView: PropTypes.func,
};

SwitchViewOptions.defaultProps = {
  currentView: GRID,
  onSwitchView: () => {},
};

export default SwitchViewOptions;
