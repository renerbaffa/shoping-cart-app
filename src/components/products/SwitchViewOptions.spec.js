import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import SwitchViewOptions from './SwitchViewOptions';
import Button from '../shared/Button';

import { GRID, LIST } from '../../constants/ViewOptions';

describe('<SwitchViewOptions />', () => {
  let wrapper;
  let onSwitchView;
  let defaultComponent;

  beforeEach(() => {
    onSwitchView = jest.fn();
    defaultComponent = (
      <SwitchViewOptions
        currentView={GRID}
        onSwitchView={onSwitchView}
        className="className"
      />
    );
    wrapper = shallow(defaultComponent);
  });

  it('should render correctly', () => {
    const switchViewOptions = renderer.create(defaultComponent).toJSON();
    expect(switchViewOptions).toMatchSnapshot();
  });

  it('should call onSwitchView prop with GRID as parameter when selecting it', () => {
    expect(onSwitchView).not.toHaveBeenCalled();
    wrapper.find(Button).at(0).simulate('click');
    expect(onSwitchView).toHaveBeenCalledWith(GRID);
  });

  it('should call onSwitchView prop with LIST as parameter when selecting it', () => {
    expect(onSwitchView).not.toHaveBeenCalled();
    wrapper.find(Button).at(1).simulate('click');
    expect(onSwitchView).toHaveBeenCalledWith(LIST);
  });

  it('should reach 100% of coverage', () => {
    SwitchViewOptions.defaultProps.onSwitchView();
  });
});
