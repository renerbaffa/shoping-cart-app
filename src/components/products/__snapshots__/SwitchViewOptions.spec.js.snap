// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SwitchViewOptions /> should render correctly 1`] = `
<div
  className="className"
>
  <button
    className="SwitchViewOptions-button Button Button-selected"
    onClick={[Function]}
  >
    Grid
  </button>
  <button
    className="SwitchViewOptions-button Button"
    onClick={[Function]}
  >
    List
  </button>
</div>
`;
